### PR TITLE
TEP-0031: Tekton Bundles CLI

### PR DESCRIPTION
**Pull request by pierretasci**
This TEP proposes additions to the Tekton CLI to support building and managing Tekton Bundles.

### CODE REVIEW

**Review by afrittoli — COMMENTED** [review]
Thank you! 
/lgtm

**Review by danielhelfand — COMMENTED** [review]
A question I have as part of this proposal, which I realize may not have come up as much in the past with TEPs, is should we have a proposed UX as part of the proposal? Or would that be considered an implementation detail that is out of scope for TEPs?

**Comment by pierretasci** [comment]
> A question I have as part of this proposal, which I realize may not have come up as much in the past with TEPs, is should we have a proposed UX as part of the proposal? Or would that be considered an implementation detail that is out of scope for TEPs?

I can't speak to the question of "should" but I can say that I was planning on adding in my proposed set of commands and what they do at a high level once I got this in and started on the "design" portion of the TEP.

**Comment by danielhelfand** [comment]
> I can't speak to the question of "should" but I can say that I was planning on adding in my proposed set of commands and what they do at a high level once I got this in and started on the "design" portion of the TEP.

Perfect, I think that makes sense. I'll take a further look at the proposal later today.

File: teps/0031-tekton-bundles-cli.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Provide a way for users to...
+1. generate valid Tekton Bundles using the Tekton CLI.
+2. fetch Tekton Bundles that they have access to.
+3. publish Tekton Bundles to the repository of their choice.
+4. inspect the contents of a Tekton Bundle.
+5. modify any of the contents of a Tekton Bundle locally, and to republish any modified Bundle.
+
+### Non-Goals
+
+- Provide an API for interacting with Tekton Bundles that users can extend. This is meant to be self-contained through
```
Comment (danielhelfand) [inline teps/0031-tekton-bundles-cli.md:59]:
```suggestion
- Provide an API for interacting with Tekton Bundles that users can extend. This is meant to be self-contained though
```

**Review by danielhelfand — COMMENTED** [review]
/lgtm

Just a minor nit.

File: teps/0031-tekton-bundles-cli.md (hunk 1 above)
Comment (pierretasci) [reply teps/0031-tekton-bundles-cli.md]: Good catch. Done.

**Comment by chmouel** [comment]
/lgtm

cc @piyush-garg @pradeepitm12 @vdemeester

**Review by pradeepitm12 — COMMENTED** [review]
This is interesting.
/lgtm

**Review by vdemeester — APPROVED** [review]

**Pull request merged**